backend/src/services/BuildingService.js

```diff
@@ -57,4 +57,10 @@ export class BuildingService extends ResourceSpender {
         return { buildings };
     };
 
+    async getByBuildingId({ kingdomId, buildingId }) {
+        if (!kingdomId) throw new Error(this.errorCodes.missingKingdomId);
+        const buildings = (await this.building.getByKingdomId({kingdomId})).filter( e => e.id == buildingId );
+        return { buildings };
+    };
```

Comment:
@gulyaasjaanos you can merge these two get
```suggestion
    async getBuilding({ kingdomId, buildingId }) {
        let buildings = (await this.building.getByKingdomId({kingdomId}))
        if(kingdomId) {
          buildings = buildings.filter( e => e.id == buildingId );
        }
        return { buildings };
    };
```